### PR TITLE
[luci-interpreter] Add a feature to infer the output shape

### PR DESCRIPTION
**Pull request by shs-park**
This PR makes a new interface `getOutputTensorSize` in the luci-interpreter.
This is to get the size of actual output size from the `_runtime_module`, which has the inferred and calculated output shape during the interprete process.

---

for issue #12979
from draft #12981

### CODE REVIEW

File: compiler/circle-interpreter/src/CircleInterpreter.cpp [hunk 1]
```diff
@@ -136,7 +136,8 @@ int entry(int argc, char **argv)
   for (int i = 0; i < module->graph()->outputs()->size(); i++)
   {
     const auto *output_node = loco::must_cast<const luci::CircleOutput *>(output_nodes[i]);
-    std::vector<char> output_data(getTensorSize(output_node));
+    size_t output_size = interpreter.getOutputTensorSize(output_node);
+    std::vector<char> output_data(output_size);
```
Comment (shs-park) [inline compiler/circle-interpreter/src/CircleInterpreter.cpp:140]: Change to get the size of output_node from the runtime module, not from the shape of origin model.
The runtime module can have inferred shape which is determined in interpreting process.

**Comment by seanshpark** [comment]
plz split `circle-interpreter` and `luci-interpreter` changes to separate PRs.

**Comment by shs-park** [comment]
> plz split `circle-interpreter` and `luci-interpreter` changes to separate PRs.

@seanshpark,
Now the `luci-interpreter` part was extracted.

File: compiler/luci-interpreter/src/Interpreter.cpp [hunk 2]
```diff
@@ -125,6 +125,21 @@ void Interpreter::readOutputTensor(const luci::CircleOutput *output_node, void *
     tensor->readData(data, data_size);
 }
 
+size_t Interpreter::getOutputTensorSize(const luci::CircleOutput *output_node)
+{
+  Tensor *tensor = _runtime_module->getOutputTensors()[output_node->index()];
+  if (tensor == nullptr)
+  {
+    const std::string &name = output_node->name();
+    throw std::runtime_error("Cannot find tensor size for output node named \"" + name + "\".");
```
Comment (seanshpark) [inline compiler/luci-interpreter/src/Interpreter.cpp:134]: NOTE other methods in this file uses `throw std::runtime_error()` for failure exit.

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by zetwhite — APPROVED** [review]
LGTM :+1:

**Comment by jinevening** [comment]
PTAL https://github.com/Samsung/ONE/pull/12981/files#r1598051074

**Review by jinevening — APPROVED** [review]
LGTM w/ minor comment (https://github.com/Samsung/ONE/pull/12981/files#r1598051074).

**Pull request merged**